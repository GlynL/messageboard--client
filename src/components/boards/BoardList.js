import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import * as actions from "../../actions/boards";
import "../../styles/boards-list.css";

class BoardList extends Component {
  componentDidMount() {
    this.props.fetchBoards();
  }

  renderBoards() {
    return (
      <ul className="boards-list__list">
        {this.props.boards.boards.map(board => (
          <li className="boards-list__list-item" key={board._id}>
            <Link className="boards-list__list-item-link" to={`/${board.name}`}>
              {board.name}
            </Link>
          </li>
        ))}
      </ul>
    );
  }

  render() {
    return <div className="boards-list">{this.renderBoards()}</div>;
  }
}

const mapStateToProps = state => ({ boards: state.boards });

export default connect(
  mapStateToProps,
  actions
)(BoardList);
