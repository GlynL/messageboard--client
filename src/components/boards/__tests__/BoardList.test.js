import React from 'react';
import { shallow } from 'enzyme';
import { BoardList } from '../BoardList';

const fetchBoards = jest.fn();

it('matches snapshot with boards', () => {
  const wrapper = shallow(<BoardList fetchBoards={fetchBoards} boards={{boards: [{_id: 1, name: 'board 1'}, {_id: 2, name: 'board 2'}]}} />);
  expect(wrapper).toMatchSnapshot();
})

it('calls fetchBoards in componentDidMount', () => {
  const wrapper = shallow(<BoardList fetchBoards={fetchBoards} boards={{boards: []}} />);
  expect(fetchBoards).toHaveBeenCalled();
})