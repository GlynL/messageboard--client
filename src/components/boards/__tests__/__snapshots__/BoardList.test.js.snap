// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with boards 1`] = `
<div
  className="boards-list"
>
  <ul
    className="boards-list__list"
  >
    <li
      className="boards-list__list-item"
      key="1"
    >
      <Link
        className="boards-list__list-item-link"
        replace={false}
        to="/board 1"
      >
        board 1
      </Link>
    </li>
    <li
      className="boards-list__list-item"
      key="2"
    >
      <Link
        className="boards-list__list-item-link"
        replace={false}
        to="/board 2"
      >
        board 2
      </Link>
    </li>
  </ul>
</div>
`;
